tests/utils.py

```diff
@@ -38,11 +42,9 @@ def render_template(template, **kwargs):
     """
     Render a Django Template
     """
-    return Template(template).render(Context(kwargs))
-
-
-def parse_template(template, **kwargs):
-    return parse_html(render_template(template, **kwargs))
+    return soup(
+        Template(template).render(Context(kwargs)), "html.parser"
+    ).prettify(formatter="html")
```

Comment:
**Issue:** Related to the comment about `djhtml`; we should not apply any formatting here, because we want to ensure formatting is consistent when editors modify the files too (although they shouldn't, let's still account for it as it's easy).

i.e. have `djhtml` configured with pre-commit, and ensure the code style is defined in `.editorconfig` so all linters (and editors) behave consistently.

Reply:
I think we do need to format the HTML output here. The snapshots are generated from this output. Our tests have `assert render_template(form) == snapshot_html`. We could move the formatting out of `render_template`, but then we'd need to update all the tests to something like `assert format_html(render_template(form)) == snapshot_html`.

Without the formatting, the rendered templates weren't very readable. Readability should be a goal for the snapshots, as they will need to be periodically reviewed.

`djhtml` doesn't format, it only indents, and some rendered output I saw had multiple nested HTML elements with no line-breaks, so we need to use something that formats. If you want to enforce a particular style other than what `bs4` spits out (indent=2), maybe we should look at `djLint`?

Reply:
As we have HTML files, we should be linting (incl. formatting) them - whether that's via `djhtml`, `djlint`, `prettier`, or something else. We should be doing this as part of pre-commit and within CI. 

As soon as we do that (currently, we exclude *.html from prettier, as it does a terrible job of HTML files with Django), then this is going to fight with the formatting applied here. While we could exclude the snapshot dirs as you say, consistently formatting/linting all HTML within the project feels neater.

This isn't a blocker for this MR as we're not yet doing any of the above, though its something to consider going forward.